### ISSUE TITLE
Problems compiling SeisSol/PUML gcc5 and gcc6.3

### ISSUE THREAD

**Issue by fvella** [SeisSol/SeisSol#38]
I am trying to compile PULM.
I got several errors:
```
src/Geometry/PUMLReader.cpp:260:101: error: array subscript has type ‘char’ [-Werror=char-subscripts]
    m_elements[cellIds[0]].neighborSides[FACE_PUML2SEISSOL[side]] = FACE_PUML2SEISSOL[ghostSide[k][i]];
```
```
submodules/PUML/PartitionMetis.h:73:26: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   for (unsigned i = 0; i < procs; i++) {
                        ~~^~~~~~~
submodules/PUML/PartitionMetis.h:83:30: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   for (unsigned int i = 0; i < m_numCells; i++) {
                            ~~^~~~~~~~~~~~
submodules/PUML/PartitionMetis.h:117:30: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
   for (unsigned int i = 0; i < m_numCells; i++)
                            ~~^~~~~~~~~~~~
```
```
src/Geometry/PUMLReader.cpp:264:64:   required from here
submodules/PUML/Downward.h:79:93: error: dependent-name ‘PUML::PUML<Topo>::vertex_t’ is parsed as a non-type, but instantiation yields a type
   internal::Utils::l2g<Topo, PUML<Topo>::vertex_t, internal::Topology<Topo>::cellvertices()>(puml, lid, gid);
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
submodules/PUML/Downward.h:79:93: note: say ‘typename PUML::PUML<Topo>::vertex_t’ if a type is meant
submodules/PUML/Downward.h:79:93: error: no matching function for call to ‘PUML::internal::Utils::l2g(const PUML::PUML<(PUML::TopoType)0u>&, unsigned int [4], long unsi
gned int*&)’
``` 
Following my enviroment
```
scons                                                  3.0.0           64bits,host-os-linux-64,scons,target-os-linux-64,tool,v3,v3.0,v3.0.0
python                                                 2.7.12          64bits,compiler,host-os-linux-64,lang-python,python,target-os-linux-64,v2,v2.7,v2.7.12
cmake                                                  3.5.1           64bits,cmake,host-os-linux-64,target-os-linux-64,tool,v3,v3.5,v3.5.1
ParMetis library                                       4.0.3           64bits,host-os-linux-64,lib,parmetis,target-os-linux-64,v4,v4.0,v4.0.3
NetCDF library                                         4.4.1           64bits,compiled-by-gcc,compiled-by-gcc-5.4.1,host-os-linux-64,lib,netcdf,target-os-linux-64,v4,v4.4,v4.4.1,vmaster
Metis library                                          5.1.0           64bits,compiled-by-gcc,compiled-by-gcc-5.4.1,host-os-linux-64,lib,metis,target-os-linux-64,v5,v5.1,v5.1.0,vmaster
MPI library                                            1.10.3          64bits,channel-stable,compiled-by-gcc,compiled-by-XSMM library                                     master-607a502  64bits,compiled-by-gcc,compiled-by-gcc-HDF5 library (static) (parallel, mpi, fortran, static) 1.10.1          64bits,channel-stable,compiled-by-gcc,compiled-by-gcc-5.4.1,hdf5,host-os-linux-64,lib,target-os-linux-64,v1,v1.10,v1.10.1,vfortran,vmpi,vparallel,vstatic
GNU C compiler                                         5.4.1           64bits,compiler,gcc,host-os-linux-64,lang-c,lang-cpp,target-os-linux-64,v5,v5.4,v5.4.1
```

**Comment by fvella** [comment]
I relaxed the how errors are treated by gcc  by addign to SConstruct:

```
-env.Append(CFLAGS   = ['-Wall', '-Werror', '-ansi'],
-           CXXFLAGS = ['-Wall', '-Werror', '-ansi'])
+env.Append(CFLAGS   = ['-Wall','-Wno-error',  '-ansi'],
+           CXXFLAGS = ['-Wall', '-Wno-error','-ansi'])
```
However I still have the following erros in PUML
```
src/Geometry/PUMLReader.cpp:264:64:   required from here
submodules/PUML/Downward.h:79:93: error: dependent-name ‘PUML::PUML<Topo>::vertex_t’ is parsed as a non-type, but instantiation yields a type
 2g<Topo, PUML<Topo>::vertex_t, internal::Topology<Topo>::cellvertices()>(puml, lid, gid);
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
submodules/PUML/Downward.h:79:93: note: say ‘typename PUML::PUML<Topo>::vertex_t’ if a type is meant
submodules/PUML/Downward.h:79:93: error: no matching function for call to ‘PUML::internal::Utils::l2g(const PUML::PUML<(PUML::TopoType)0u>&, unsigned int [4], long unsi
gned int*&)’
```

**Comment by fvella** [comment]
We fixed by applying the following patch:

```
diff --git a/Downward.h b/Downward.h
index 591c95f..01d38db 100644
--- a/Downward.h
+++ b/Downward.h
@@ -76,7 +76,7 @@ public:
        {
                unsigned int lid[internal::Topology<Topo>::cellvertices()];
                vertices(puml, cell, lid);
-               internal::Utils::l2g<Topo, PUML<Topo>::vertex_t, internal::Topology<Topo>::cellvertices()>(puml, lid, gid);
+               internal::Utils::l2g<Topo, typename PUML<Topo>::vertex_t, internal::Topology<Topo>::cellvertices()>(puml, lid, gid);
        }
 
        /**
```

**Comment by uphoffc** [comment]
Thanks for posting a fix. I've included it and also fixes for the signed/unsigned warnings in c44dc29.
